### PR TITLE
Convert status to not be a pointer in the Span

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: consumer/pdata/trace.go [hunk 1]
```diff
@@ -168,7 +168,7 @@ func (sc StatusCode) String() string { return otlptrace.Status_StatusCode(sc).St
 //
 // Important: This causes a runtime error if IsNil() returns "true".
 func (ms SpanStatus) SetCode(v StatusCode) {
```
Comment (tigrannajaryan) [inline consumer/pdata/trace.go:170]: >// Important: This causes a runtime error if IsNil() returns "true".

Do we still have IsNil() for this struct?

File: receiver/opencensusreceiver/opencensus_test.go [hunk 2]
```diff
@@ -153,7 +153,7 @@ func TestGrpcGateway_endToEnd(t *testing.T) {
 	assert.True(t, proto.Equal(want.Resource, gotOc[0].Resource))
 	require.Len(t, want.Spans, 1)
 	require.Len(t, gotOc[0].Spans, 1)
-	assert.True(t, proto.Equal(want.Spans[0], gotOc[0].Spans[0]))
+	assert.EqualValues(t, want.Spans[0], gotOc[0].Spans[0])
```
Comment (tigrannajaryan) [inline receiver/opencensusreceiver/opencensus_test.go:156]: Why was this change needed?

File: translator/internaldata/traces_to_oc.go [hunk 3]
```diff
@@ -386,10 +386,6 @@ func spanIDToOC(sid pdata.SpanID) []byte {
 }
 
 func statusToOC(status pdata.SpanStatus) (*octrace.Status, *octrace.AttributeValue) {
-	if status.IsNil() {
-		return nil, nil
-	}
-
```
Comment (tigrannajaryan) [inline translator/internaldata/traces_to_oc.go:392]: Should we return (nil, nil) if `status.Code()==case pdata.StatusCodeUnset && status.message==""`?

File: receiver/opencensusreceiver/opencensus_test.go (hunk 2 above)
Comment (bogdandrutu) [reply receiver/opencensusreceiver/opencensus_test.go:156]: Because this shows diffs between them in the failure, not just fail true != false :)

File: translator/internaldata/traces_to_oc.go (hunk 3 above)
Comment (bogdandrutu) [reply translator/internaldata/traces_to_oc.go:392]: Not sure, was thinking about that as well. Is that not a different change? That behavior was not there before. Do you suggest to do this?

File: consumer/pdata/trace.go (hunk 1 above)
Comment (bogdandrutu) [reply consumer/pdata/trace.go]: Done.

**Review by carlosalberto — APPROVED** [review]

File: translator/internaldata/traces_to_oc.go (hunk 3 above)
Comment (tigrannajaryan) [reply translator/internaldata/traces_to_oc.go:392]: Because we now don't have a pointer to Status this case will happen always for all successful statuses, so it is a bit more important now. But, probably doesn't matter much, it only affects OpenCensus format. Let's leave as is and we can change in the future if needed.

**Review by tigrannajaryan — APPROVED** [review]
